stories/patterns/ux-patterns/confirmation-modal-dialog/docs/examples/confirmation-modal-example.tsx

```diff
@@ -0,0 +1,171 @@
+import React from 'react';
+import Button from '@semcore/button';
+import { Flex } from '@semcore/flex-box';
+import Modal from '@semcore/modal';
+import { Text, List } from '@semcore/typography';
+import Input from '@semcore/input';
+import Tooltip from '@semcore/tooltip';
+import '@semcore/utils/lib/themes/default.css';
+import { useForm, SubmitHandler } from 'react-hook-form';
+
+const warningBlockStyles = {
+  background: 'var(--intergalactic-bg-secondary-critical)',
+  border: '1px solid var(--intergalactic-border-critical)',
+  borderRadius: 'var(--intergalactic-surface-rounded)',
+  marginBottom: '24px',
+  padding: '16px',
+};
+
+type FormValues = {
+  project: string;
+};
+const fieldName: keyof FormValues = 'project';
+
+const Demo = () => {
+  const { register, trigger, handleSubmit, reset, resetField, getValues, formState } =
+    useForm<FormValues>({
+      mode: 'onBlur',
+      defaultValues: { project: '' },
+    });
+
+  const [visible, setVisible] = React.useState(false);
+  const handleOpen = React.useCallback(() => setVisible(true), []);
+  const handleClose = React.useCallback(() => {
+    setVisible(false);
+    reset({ project: '' }, { keepIsSubmitted: false, keepTouched: false });
+  }, []);
+
+  const { errors, isSubmitted, touchedFields } = formState;
+  const [focusedFieldName, setFocusedFieldName] = React.useState('');
+  const onSubmit: SubmitHandler<FormValues> = handleClose;
+
+  const hasError = () => {
+    const error = errors[fieldName];
+
+    if (error && error.type === 'projectRequired' && !isSubmitted) {
+      return false;
+    }
+
+    return Boolean(error);
+  };
+
+  const invalid = (): boolean => {
+    return Boolean(touchedFields[fieldName]) && hasError();
+  };
+
+  const showErrorTooltip = (): boolean => {
+    const isActive = focusedFieldName === fieldName;
+    return invalid() && isActive;
+  };
+
+  const resetIfChangedToValid = (isValid: boolean) => {
+    if (isValid) {
+      resetField(fieldName, { keepTouched: false, defaultValue: getValues(fieldName) });
+    }
+  };
+
+  type RegisterOptions = Parameters<typeof register>[1];
```

Comment:
let's move type at the top, out of function

Reply:
It seems to me logical to keep type and function together as both are closely related and used only together. If this does not contradicts common project practices, maybe we could keep it here?